community/CM-Configuration-Management/policy-create-ns-backup-triliovault-for-kubernetes-templatized.yaml

```
@@ -0,0 +1,200 @@
+# This policy creates a namespace based backup using Triliovault for Kubernetes (TVK)
+# on all OpenShift managed clusters with a label "protected-by=triliovault".
+#
+# This policy creates backup for all the namespaces with a lable "protected-by=tvk-ns-backup"
+#
+# To create the namespace backup, following process is followed
+# 1. Create the target object to store backup data
+#    This policy creates AWS S3 based target storage for storing backup images.
+# 2. Create a backup plan that references the following:
+#      Namespace to be backed up
+#      Default Scheduling Policy - create daily 1 backup
+#      Target to where the data should be backed up
+# 3. Apply the Backup CR that references the Backup
+#    This policy creates a full backup.
+#
+# IMPORTANT: Please follow below instructions for the policy to work
+# On the hub cluster, create a secret aws-s3-secret in "default" namespace which has AWS S3 details for target creation
+# example yaml (replace the values)
+# apiVersion: v1
+# kind: Secret
+# metadata:
+#   name: aws-s3-secret
+#   namespace: default
+# type: Opaque
+# stringData:
+#   accessKey: "PROVIDE_ACCESS_KEY"
+#   secretKey: "PROVIDE_SECRET_KEY"
+#   bucketName: "PROVIDE_S3_BUCKET_NAME"
+#   region: "PROVIDE_REGION"
+#   thresholdCapacity: 100Gi
+#
+apiVersion: policy.open-cluster-management.io/v1
+kind: Policy
+metadata:
+  name: policy-kyverno-tvk-create-ns-backup
+  annotations:
+    policy.open-cluster-management.io/categories: CA Security Assessment and Authorization
+    policy.open-cluster-management.io/standards: NIST SP 800-53
+    policy.open-cluster-management.io/controls: CA-2 Security Assessments, CA-7 Continuous Monitoring
+spec:
+  disabled: false
+  remediationAction: inform
+  policy-templates:
+    - objectDefinition:
+        apiVersion: policy.open-cluster-management.io/v1
+        kind: ConfigurationPolicy
+        metadata:
+          name: tvk-s3-secret
+        spec:
+          remediationAction: enforce
+          severity: high
+          object-templates:
+            - complianceType: musthave
+              objectDefinition:
+                apiVersion: v1
+                kind: Secret
+                metadata:
+                  name: tvk-s3-secret
+                  namespace: default
+                type: Opaque
+                stringData:
+                  accessKey: '{{hub fromSecret "default" "aws-s3-secret" "accessKey" hub}}'
```

Don't put in the namespace since Hub templates default to using the namespace that the policy is in and any other namespace put in will fail due to security restrictions.
```suggestion
                  accessKey: '{{hub fromSecret "" "aws-s3-secret" "accessKey" hub}}'
```

---

Got it.